src/parsers/fixed-bytes.ts

```diff
@@ -72,6 +73,10 @@ export const fixedBytes: Parser<BytesLike, Uint8Array> = {
       ),
     );
 
+    if (packed) {
+      return concatBytes([buffer, padEnd(bufferValue, length)]);
```

Comment:
Yeah, values are always padded to their length, except for dynamic values (since those don't have a length).

`uint16` is always two bytes, `bytes8` is always eight bytes, and so on.